numpy/core/src/multiarray/methods.c

```diff
@@ -2598,45 +2598,24 @@ array_complex(PyArrayObject *self, PyObject *NPY_UNUSED(args))
 
     if (!PyArray_CanCastArrayTo(self, dtype, NPY_SAME_KIND_CASTING) &&
             !(PyArray_TYPE(self) == NPY_OBJECT)) {
-        PyObject *err, *msg_part;
+        PyObject *descr = (PyObject*)PyArray_DESCR(self);
+
+        PyErr_Format(PyExc_TypeError, "Unable to convert %R to complex", descr);
         Py_DECREF(dtype);
```

Comment:
nit: move this up a line - decref things as soon as you're done with them.

Reply:
How do you know we are done with it until you read to the end?

Reply:
My motivation is to make a statement of intent, not to write cautious code. If our goal was to be cautious, then we'd use `XDECREF` everywhere.